src/main/java/org/opensearch/neuralsearch/processor/rerank/context/DocumentContextSourceFetcher.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.processor.rerank.context;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.xcontent.ObjectPath;
+import org.opensearch.search.SearchHit;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+/**
+ * Context Source Fetcher that gets context from the search results (documents)
+ */
+@Log4j2
+@AllArgsConstructor
+public class DocumentContextSourceFetcher implements ContextSourceFetcher {
+
+    public static final String NAME = "document_fields";
+    public static final String DOCUMENT_CONTEXT_LIST_FIELD = "document_context_list";
+
+    List<String> contextFields;
+
+    /**
+     * Fetch the information needed in order to rerank.
+     * That could be as simple as grabbing a field from the search request or
+     * as complicated as a lookup to some external service
+     * @param searchRequest the search query
+     * @param searchResponse the search results, in case they're relevant
+     * @param listener be async
+     */
+    @Override
+    public void fetchContext(SearchRequest searchRequest, SearchResponse searchResponse, ActionListener<Map<String, Object>> listener) {
+        List<String> contexts = new ArrayList<>();
+        for (SearchHit hit : searchResponse.getHits()) {
+            StringBuilder ctx = new StringBuilder();
+            for (String field : this.contextFields) {
+                ctx.append(contextFromSearchHit(hit, field));
+            }
+            contexts.add(ctx.toString());
+        }
+        listener.onResponse(new HashMap<>(Map.of(DOCUMENT_CONTEXT_LIST_FIELD, contexts)));
+    }
+
+    private String contextFromSearchHit(final SearchHit hit, final String field) {
+        if (hit.getFields().containsKey(field)) {
+            Object fieldValue = hit.field(field).getValue();
+            return String.valueOf(fieldValue);
+        } else if (hit.hasSource() && hit.getSourceAsMap().containsKey(field)) {
+            Object sourceValue = ObjectPath.eval(field, hit.getSourceAsMap());
+            return String.valueOf(sourceValue);
+        } else {
```

Comment:
we don't need `else` block here

Reply:
we don't, but I think it's more readable this way (at least here)

Reply:
makes sense